### PR TITLE
Support BP_WEB_SERVER_LOCATION_PATH env var

### PR DESCRIPTION
**Pull request by francescopersico**
## Summary

The PR add the support of an env var named BP_WEB_SERVER_LOCATION_PATH to support hosting application from a subpath

## Use Cases

When hosting a staticapp under a subpath is important that nginx is configured appropriately.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: default_config_generator_test.go [hunk 1]
```diff
@@ -251,6 +251,19 @@ error_log stderr;
 			Expect(string(contents)).To(ContainSubstring(`root /some/absolute/path;`))
 		})
 
+		it("writes a nginx.conf with specified location path", func() {
```
[the remark this answers is not in the document]
Comment (francescopersico) [reply default_config_generator_test.go:254]: If you want i can add it but i think it's already covered by the test "writes a default nginx.conf to the working directory" that perform the check with no env vars

In particular this check

```
Expect(string(contents)).To(ContainSubstring(`  server {
    listen {{port}} default_server;
    server_name _;

    # Directory where static files are located
    root {{ env "APP_ROOT" }}/public;

    location / {
      # Specify files sent to client if specific file not requested (e.g.
      # GET www.example.com/). NGINX sends first existing file in the list.
      index index.html index.htm Default.htm;
    }

    # (Security) Don't serve dotfiles, except .well-known/, which is needed by
    # LetsEncrypt
    location ~ /\.(?!well-known) {
      deny all;
      return 404;
    }
  }
```

as you can see there is the string "location /".

What is your opinion?

**Comment by francescopersico** [comment]
@robdimsdale Done!

**Comment by francescopersico** [comment]
@ForestEckhardt Sure! Going to check integration tests.

**Comment by ForestEckhardt** [comment]
@paketo-buildpacks/web-servers-maintainers We should look into adding the same functionality to HTTPD.

File: integration/no_conf_app_test.go [hunk 2]
```diff
@@ -69,6 +69,7 @@ func testNoConfApp(t *testing.T, context spec.G, it spec.S) {
 			Expect(logs).To(ContainLines(
 				"  Generating /workspace/nginx.conf",
 				`    Setting server root directory to '{{ env "APP_ROOT" }}/public'`,
+				"    Setting server location path to /",
```
Comment (ForestEckhardt) [inline integration/no_conf_app_test.go:72]:
```suggestion
				"    Setting server location path to '/'",
```

**Review by ForestEckhardt — APPROVED** [review]

**Comment by ForestEckhardt** [comment]
@paketo-buildpacks/web-servers-maintainers If I could get one more set of eyes on this I would appreciate it.

**Review by TisVictress — APPROVED** [review]
LGTM!

**Pull request merged**